onmt/Loss.py

```diff
@@ -121,7 +121,7 @@ def sharded_compute_loss(self, batch, output, attns,
 
         for shard in shards(shard_state, shard_size):
             loss, stats = self._compute_loss(batch, **shard)
-            loss.div(normalization).backward()
+            loss.div(normalization).backward(retain_graph=True)
```

Comment:
you need to cast float(normalization) otherwise .div() returns a long tensor instead of a Float Tensor